### DISCUSSION TITLE
Assigning Multiple Columns to a RefineableString

### DISCUSSION THREAD

**Discussion by ro2Partners** [microsoft-search/pnp-modern-search#3721] [Q&A]
I have a few document libraries (4) that I want to pull into a single search interface using pnp-modern search. All of these document libraries have two columns that use globally managed metadata. How do I ensure that the indexed metadata from each document library is added to the RefinableString that I have associated with a filter in the search? I.e., I have 1 document tagged with the same metadata across 4 document libraries. In my search, all of the documents show, but only one document document from the hub site document libraries can be filtered.    I looked for similar discussions but was not able to find any although I'm sure this has been discussed before.

**Comment by kasperbolarsen** [comment]
First of all, those managed metadata columns, are they defined in the Content Type gallery, or as site columns in the same Site as the 4 libraries reside on?

It is a prerequisite that it is the same site column in all 4 libraries if you want the mapping to be straightforward. Otherwise you will have to map several crawled properties to the same RefinableString for it to work.

**Reply by ro2Partners** [reply]
I created a site column in the main hub site under the "custom column" grouping, assuming it would roll to the associated sites, but this appears not to be the case. I couldn't find it when I tried to add the site column to the associated site. Does this apply only to document libraries on the same site or document libraries across multiple sites?

**Reply by kasperbolarsen** [reply]
Yeah, the Content type gallery is the only reliable way to sync content types across sites, unless you want to handle it yourself using code.

**Reply by ro2Partners** [reply]
@kasperbolarsen Thank you for your insightful feedback!!

**Reply by ro2Partners** [reply]
@kasperbolarsen I set up the content gallery and added the columns to another document library on a separate site. I see all of the documents returned in the search; however, I only see metadata from one document library. 

Sample query 

{searchTerms} 
Path: "https://sharepoint.com/Shared%20Documents" or
Path: "https://sharepoint.com/sites/Shared%20Documents1") or
Path: "https://sharepoint.com/sitesShared%20Documents2" 

Is there anything wrong with my KQL?

**Reply by kasperbolarsen** [reply]
If that is you query then yes. You have a rogue parentheses in the query

**Reply by ro2Partners** [reply]
@kasperbolarsen I removed the parentheses, but the results are still the same.  IDK, at this point, any other feedback would be appreciated.

**Reply by kasperbolarsen** [reply]
Feel free to send the real question to me on X ( I have same username in X as here)
